cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java

```diff
@@ -198,6 +198,13 @@ public BuildCommand() {
             "generation")
     private String graalVMBuildOptions;
 
+    @CommandLine.Option(names = "--optimize", description = "generate optimized executable jar", defaultValue = "false")
+    private Boolean optimizeCodegen;
+
+    @CommandLine.Option(names = "--optimize-report", description = "generate codegen optimization reports",
```

Comment:
Aren't we adding this to RunCommand also?

Reply:
Users expect the same behavior for `bal run` and `bal build` + running the executable. So, I proposed this to maintain consistency. @sameerajayasoma WDYT?

Reply:
As per the offline discussion with @hasithaa, we decided not to add support for the `bal run` command because. This was because this project's main goal is to reduce the graalvm build time. And graalvm native image build cannot be done through `bal run` command.